### DISCUSSION TITLE
GPLVM without specifying n (or GPLVM for very large datasets)

### DISCUSSION THREAD

**Discussion by ManjunathSripadarao** [cornellius-gp/gpytorch#1852] [Q&A]
Hello,
I was looking at the GPLVM example [here](https://docs.gpytorch.ai/en/latest/examples/045_GPLVM/Gaussian_Process_Latent_Variable_Models_with_Stochastic_Variational_Inference.html) and in the case of very large datasets, n is very large.

```
class bGPLVM(BayesianGPLVM):
    def __init__(self, n, data_dim, latent_dim, n_inducing, pca=False):
        self.n = n
        self.batch_shape = torch.Size([data_dim])
        ...
        # Locations Z_{d} corresponding to u_{d}, they can be randomly initialized or
        # regularly placed with shape (D x n_inducing x latent_dim).
        self.inducing_inputs = torch.randn(data_dim, n_inducing, latent_dim)     # and here ?
        ...
        # Define prior for X
        X_prior_mean = torch.zeros(n, latent_dim)  # shape: N x Q <- Here
        prior_x = NormalPrior(X_prior_mean, torch.ones_like(X_prior_mean))

```

Is it possible to setup a model where the n need not be specified ? As the data streams in the model is built ? Or some sort of a sparse model ? I have a dataset that runs into 100s of GBs and it has each datapoint being 480-520 dimensions and I want to compress it to 64 dimensions (or lesser). Also my batch size is around 2048 if that makes any difference (I can go lower too).

@vr308 @gpleiss ?

Sorry if this question is already answered or is irrelevant. One requirement is that we want to use a GPLVM or a GP model for this, not a neural network. Something where we can characterize uncertainty...

Thanks in advance.

**Comment by vr308** [comment]
Hi @ManjunathSripadarao you can replace the line :

`self.inducing_inputs = torch.randn(data_dim, n_inducing, latent_dim)     # and here ?`
with 
`self.inducing_inputs = torch.randn(n_inducing, latent_dim)  ## sharing inducing inputs across all dimensions`

that should make a difference?
